datasketches/src/frequencies/serde.rs

```diff
@@ -0,0 +1,141 @@
+            out.extend_from_slice(&len.to_le_bytes());
+            out.extend_from_slice(bytes);
+        }
+        out
+    }
+
+    fn deserialize_items(
+        &self,
+        bytes: &[u8],
+        num_items: usize,
+    ) -> Result<(Vec<String>, usize), SerdeError> {
+        if num_items == 0 {
+            return Ok((Vec::new(), 0));
+        }
+        let mut items = Vec::with_capacity(num_items);
+        let mut offset = 0usize;
+        for _ in 0..num_items {
+            if offset + 4 > bytes.len() {
+                return Err(SerdeError::InsufficientData(
+                    "not enough bytes for string length".to_string(),
+                ));
+            }
+            let len = u32::from_le_bytes([
+                bytes[offset],
+                bytes[offset + 1],
+                bytes[offset + 2],
+                bytes[offset + 3],
+            ]) as usize;
+            offset += 4;
+            if offset + len > bytes.len() {
+                return Err(SerdeError::InsufficientData(
+                    "not enough bytes for string payload".to_string(),
+                ));
+            }
+            let slice = &bytes[offset..offset + len];
+            let value = match str::from_utf8(slice) {
+                Ok(s) => s.to_string(),
+                Err(_) => {
+                    return Err(SerdeError::MalformedData(
+                        "invalid UTF-8 string payload".to_string(),
+                    ));
+                }
+            };
+            items.push(value);
+            offset += len;
+        }
+        Ok((items, offset))
+    }
+}
+
+/// Serializer for i64 items compatible with ArrayOfLongsSerDe in Java.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct I64Serde;
+
+impl ItemsSerde<i64> for I64Serde {
+    fn serialize_items(&self, items: &[i64]) -> Vec<u8> {
+        let mut out = Vec::with_capacity(items.len() * 8);
+        for item in items {
+            out.extend_from_slice(&item.to_le_bytes());
+        }
+        out
+    }
+
+    fn deserialize_items(
+        &self,
+        bytes: &[u8],
+        num_items: usize,
+    ) -> Result<(Vec<i64>, usize), SerdeError> {
+        let needed = num_items
+            .checked_mul(8)
+            .ok_or_else(|| SerdeError::MalformedData("items size overflow".to_string()))?;
+        if bytes.len() < needed {
+            return Err(SerdeError::InsufficientData(
+                "not enough bytes for i64 items".to_string(),
+            ));
+        }
+        let mut items = Vec::with_capacity(num_items);
+        for i in 0..num_items {
+            let offset = i * 8;
+            let value = i64::from_le_bytes([
+                bytes[offset],
+                bytes[offset + 1],
+                bytes[offset + 2],
+                bytes[offset + 3],
+                bytes[offset + 4],
+                bytes[offset + 5],
+                bytes[offset + 6],
+                bytes[offset + 7],
+            ]);
```

Comment:
```suggestion
use crate::frequencies::serialization::read_i64_le;
let value = read_i64_le(bytes, offset);
```

Reduce duplicate.

BTW, this may be later refactored with https://github.com/apache/datasketches-rust/issues/41.